mhs-reference-implementation/mhs/sender/sender.py

```diff
@@ -22,54 +21,80 @@ def __init__(self, interaction_name):
 
 
 class Sender:
-    def __init__(self, interactions_config, message_builder, transport):
+    def __init__(self, interactions_config, message_builder, transport, party_id):
         """Create a new Sender that uses the specified dependencies to load config, build a message and send it.
 
         :param interactions_config: The object that can be used to obtain interaction-specific configuration details.
         :param message_builder: The message builder that can be used to build a message.
         :param transport: The transport that can be used to send messages.
+        :param party_id: The party ID of this MHS. Sent in ebXML requests.
         """
 
         self.interactions_config = interactions_config
         self.message_builder = message_builder
         self.transport = transport
+        self.party_id = party_id
 
-    def send_message(self, interaction_name, message_to_send):
-        """Send the specified message for the interaction named.
+    def build_message(self, interaction_name: str, content: str) -> Tuple[str, str]:
```

Comment:
Just a little confused. We have a `message_builder` that we use to build a message but then we also have this. Do you think it might be worth clearing as the difference here? From what I understand this looks like a wrapper and the actual message inside here is the one that the message_builder builds?

Reply:
As discussed offline, I'll update this method to be `prepare_message` and make it clearer in the docstring that it's preparing the message received for sending.